synapse/handlers/auth.py

```diff
@@ -1435,20 +1435,24 @@ async def delete_access_token(self, access_token: str) -> None:
             access_token: access token to be deleted
 
         """
-        user_info = await self.auth.get_user_by_access_token(access_token)
+        token = await self.store.get_user_by_access_token(access_token)
+        if not token:
+            # At this point, the token should already have been fetched once by
+            # the caller, so this should not happen
+            raise SynapseError(HTTPStatus.UNAUTHORIZED, "Unrecognised access token")
```

Comment:
It's a bit weird, but everywhere where `delete_access_token` is called, we're already checking if the token exists. But since `delete_access_token` is kinda really tied to tokens in the database (it does not make sense to delete a guest access token, since it can't be invalidated), I changed this to use the store's `get_user_by_access_token` instead of the auth one, and this one does not raise when a token does not exist. So this is mainly there to appease mypy

Reply:
> So this is mainly there to appease mypy

It's presumably possible for two delete requests to race against one another, in which case one of them could actually see `token is None` here?

Suggest updating the "this should not happen" comment, assuming you agree

Reply:
Done